sklearn/utils/stats.py

```diff
@@ -7,11 +7,14 @@
 )
 
 
-def _weighted_percentile(array, sample_weight, percentile_rank=50, xp=None):
-    """Compute the weighted percentile with method 'inverted_cdf'.
+def _weighted_percentile(
+    array, sample_weight, percentile_rank=50, average=False, xp=None
+):
+    """Compute the weighted percentile.
 
-    When the percentile lies between two data points of `array`, the function returns
-    the lower value.
```

Comment:
I thought about replacing this sentence with a description of how `inverted_cdf` and `averaged_inverted_cdf` behaves but I sort of think the descriptions in the paper with the equations are clearer.

I also think it's initially 'confusing' when I describe that when a percentile falls exactly on a value in the array, we take the average of it and the next value in the array. It makes sense when you think about an array of 4 values, 50 percentile would fall exactly on the 2nd value, but of course symmetrized it makes sense to take the average of the 2nd and 3rd values.

I am happy to add this still if you think it is appropriate.

Reply:
Yes, please feel free to go ahead.